src/framework-ondata.js

```diff
@@ -35,18 +35,21 @@ module.exports = function(payload, socketIndex) {
         nodeInstance: this.nodeInstance
     });
 
-     var vnid = payload.vnid || '';
-     var vni = this.nodeInstance.vni(vnid);
+    var vnid = payload.vnid || '';
+    var vni = this.nodeInstance.vni(vnid);
 
-     // Get the old & new input state for port and the new payload
-     var lastInputState = vni.inputStates(portName, socketIndex);
-     var inputState = (_.isUndefined(payload.vnid)) ? createState(vnid, payload) : payload;
+    // Get the old & new input state for port and the new payload
+    var lastInputState = vni.inputStates(portName, socketIndex);
+    var inputState = (_.isUndefined(payload.vnid)) ? createState(vnid, payload) : payload;
  
-     // Check if it's stale and if so, hande setting output flag and sending it on downstream if appropriate
-     var isStale = handleStaleData(vni, lastInputState, inputState, outputPorts.output);
+    // Check if it's stale and if so, hande setting output flag and sending it on downstream if appropriate
+    var isStale = handleStaleData(vni, lastInputState, inputState, outputPorts.output);
 
-     // Set the new input state
-     vni.inputStates(portName, socketIndex, inputState);
+    // Set the new input state
+    vni.inputStates(portName, socketIndex, inputState);
+
+    // append any input state metadata to the outputState so it gets passed on to downstream nodes
+    transmitMetadata(vni, inputState);
```

Comment:
Suggestions: 

1. s/transmitMetadata/setDefaultMetadata/ 

2. Would it make sense to move this call inside runUpdater, next to the call of clearStateData, so that it is only called if the updater will actually be run?